android/feature/find-meal/src/main/java/com/kinandcarta/feature/find/meal/view/FindMealFragment.kt

```diff
@@ -1,67 +1,113 @@
 package com.kinandcarta.feature.find.meal.view
 
+import android.annotation.SuppressLint
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.FrameLayout
 import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.viewModels
-import androidx.recyclerview.widget.LinearLayoutManager
-import androidx.recyclerview.widget.RecyclerView
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.SupportMapFragment
 import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MarkerOptions
+import com.google.android.material.bottomsheet.BottomSheetBehavior
 import com.kinandcarta.feature.find.meal.R
 import com.kinandcarta.feature.find.meal.adapter.MealsAdapter
 import com.kinandcarta.feature.find.meal.databinding.FindMealFragmentBinding
-import com.kinandcarta.feature.find.meal.extension.showToast
+import com.kinandcarta.feature.find.meal.extension.*
+import com.kinandcarta.feature.find.meal.ui.BottomSheetBehaviourCallback
 import com.kinandcarta.feature.find.meal.viewmodel.DisplayMealsViewModel
 import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
 
 @AndroidEntryPoint
 class FindMealFragment : Fragment() {
 
-    private val progressBarView: FrameLayout by lazy { binding.progressBar }
     private val viewModel: DisplayMealsViewModel by viewModels()
-    private val mealsRecyclerView: RecyclerView by lazy { binding.rvMeals }
-    private val mealsAdapter = MealsAdapter { id -> viewModel.reserveAMeal(id) }
+
     private var _binding: FindMealFragmentBinding? = null
     private val binding get() = _binding!!
```

Comment:
What does it mean the double question mark for a Kotlin point of view? As a Swift developer, I feel very scary to see   this.

Reply:
It literally means: this nullable variable _**won't**_ be `null` at the time it's accessed. 

As a Kotlin developer, I am also very afraid of seeing `!!` hahaha. As a general rule of thumb, I'd decline any PR containing `!!`, but this case is well justified, let me explain why:

When a Fragment calls `onCreateView` we use ViewBinding to get the layout, but when the Fragment calls `onDestroyView` we **_must_** destroy that layout reference. (This is one of the changes a few lines below this).
In other words, we can guarantee that `_binding` won't be null between `onCreateView` and `onDestroyView`.

IMO it's a bit ugly, but it's the only recommended way of dealing with `ViewBinding` according to the [documentation](https://developer.android.com/topic/libraries/view-binding#fragments).